# FORTUNE 500 COMPANIES - 2020 : Insights from Data

The Fortune 500 is an annual list compiled and published by Fortune magazine that ranks 500 of the largest United States corporations by the annual total revenue using publicly available data.

The Fortune 500, created by Edgar P. Smith, was first published in 1955.





In [3]:
import numpy as np 
import pandas as pd 

In [5]:
df = pd.read_csv('Fortune 2020.csv')


## Top 10 Fortune 500 Companies

In [6]:
df[:10]

rank             company  ... Assets(millions)  market_value
0     1             Walmart  ...       236,495.00    321,803.30
1     2          Amazon.com  ...       225,248.00    970,680.10
2     3         Exxon Mobil  ...       362,597.00    160,696.30
3     4               Apple  ...       338,516.00  1,112,640.80
4     5          CVS Health  ...       222,449.00     77,375.80
5     6  Berkshire Hathaway  ...       817,729.00    442,897.10
6     7  UnitedHealth Group  ...       173,889.00    236,555.20
7     8            McKesson  ...        59,672.00     21,845.00
8     9                AT&T  ...       551,669.00    209,387.50
9    10   AmerisourceBergen  ...        39,172.00     18,221.40

[10 rows x 10 columns]

In [7]:
df.shape

(500, 10)


There are 500 Companies in this table with 10 Columns in which 

> 1st column indicates Rank , 

> 2nd column is dedicated to the Company name and 

> The rest 8 columns provide information about each company


In [9]:
for i in df.columns:
  print(i, ' : ', df[i].dtype)

rank  :  int64
company  :  object
no_of_employees  :  object
rank_change  :  object
revenues($millions)  :  object
revenue_change  :  object
profits($millions)  :  object
profit_change  :  object
Assets(millions)  :  object
market_value  :  object


### Datatype Conversion : From object to float


In [16]:
df = df.replace('-','0')

In [11]:
def remove_comma(x):
  x = x.replace(',','')
  return x

In [13]:
cols_comma_remove = ['no_of_employees','revenues($millions)','profits($millions)','Assets(millions)','market_value' ]
for i in cols_comma_remove:
  df[i] = df[i].apply(remove_comma)

In [17]:
for i in df.columns[2:]:
  df[i] = df[i].astype(float)
for i in df.columns:
  print(i, ' : ', df[i].dtype)

rank  :  int64
company  :  object
no_of_employees  :  float64
rank_change  :  float64
revenues($millions)  :  float64
revenue_change  :  float64
profits($millions)  :  float64
profit_change  :  float64
Assets(millions)  :  float64
market_value  :  float64


## Rank Analysis : 

In [19]:
df['rank_is_changed'] = df['rank_change'].apply(lambda x : False if x==0.0 else True)
rank_change_df = df[df['rank_is_changed']== False]
rank_change_df.reset_index(drop=True, inplace=True)
rank_change_df.head()

rank            company  ...  market_value  rank_is_changed
0     1            Walmart  ...      321803.3            False
1     9               AT&T  ...      209387.5            False
2    10  AmerisourceBergen  ...       18221.4            False
3    12         Ford Motor  ...       19150.6            False
4    14   Costco Wholesale  ...      125907.7            False

[5 rows x 11 columns]

In [21]:
rank_increased_companies = df[df['rank_change']>0.0]
rank_increased_companies.reset_index(drop=True, inplace=True)
rank_increased_companies.head()

rank         company  ...  market_value  rank_is_changed
0     2      Amazon.com  ...      970680.1             True
1     5      CVS Health  ...       77375.8             True
2    11        Alphabet  ...      798905.3             True
3    13           Cigna  ...       65896.7             True
4    17  JPMorgan Chase  ...      276750.1             True

[5 rows x 11 columns]

In [22]:
rank_decreased_companies = df[df['rank_change']<0.0]
rank_decreased_companies.reset_index(drop=True, inplace=True)
rank_decreased_companies.head()

rank             company  ...  market_value  rank_is_changed
0     3         Exxon Mobil  ...      160696.3             True
1     4               Apple  ...     1112640.8             True
2     6  Berkshire Hathaway  ...      442897.1             True
3     7  UnitedHealth Group  ...      236555.2             True
4     8            McKesson  ...       21845.0             True

[5 rows x 11 columns]

## Top 10 Companies of the Rank Analysis Table 

In [23]:
no_change_list = rank_change_df['company'].tolist()
inc_list = rank_increased_companies['company'].to_list()
dec_list = rank_decreased_companies['company'].to_list()
top10_no_change = no_change_list[:10]
top10_inc = inc_list[:10]
top10_dec = dec_list[:10]
rank_data = {'Rank_Retained' : top10_no_change,
           'Rank_Increased' : top10_inc,
           'Rank_Decreased' : top10_dec
           }
rank_df = pd.DataFrame(rank_data)
rank_df[:10]

Rank_Retained      Rank_Increased            Rank_Decreased
0               Walmart          Amazon.com               Exxon Mobil
1                  AT&T          CVS Health                     Apple
2     AmerisourceBergen            Alphabet        Berkshire Hathaway
3            Ford Motor               Cigna        UnitedHealth Group
4      Costco Wholesale      JPMorgan Chase                  McKesson
5       Cardinal Health           Microsoft                   Chevron
6       Bank of America  Marathon Petroleum            General Motors
7     Dell Technologies          Home Depot  Walgreens Boots Alliance
8  State Farm Insurance             Comcast    Verizon Communications
9                   IBM              Anthem                    Kroger

In [25]:
rank_df.to_csv('Rank Analysis.csv')

In [49]:
rank_analysis = [rank_change_df, rank_increased_companies, rank_decreased_companies]
for i in rank_analysis:
  print(len(i.index))

44
236
220


### No of companies which retained their Rank : 44

### No of companies which improved their Rank : 236

### No of companies whose Rank dropped : 220

## Profit Analysis:

In [44]:
profit_df = df.sort_values(by = ['profits($millions)'], ascending = False, ignore_index=True)
profit_df[['rank','company', 'profits($millions)']][:10]

rank                 company  profits($millions)
0     6      Berkshire Hathaway             81417.0
1     4                   Apple             55256.0
2    21               Microsoft             39240.0
3    17          JPMorgan Chase             36431.0
4    11                Alphabet             34343.0
5    25         Bank of America             27430.0
6    45                   Intel             21048.0
7    30             Wells Fargo             19549.0
8    31               Citigroup             19401.0
9    20  Verizon Communications             19265.0

In [45]:
profit_change_df = df.sort_values(by = ['profit_change'], ascending = False, ignore_index=True)
profit_change_df[['rank','company','profit_change']][:10]

rank                              company  profit_change
0    63                        Cisco Systems         104.65
1   319                              Calpine          76.00
2   351                            Alleghany          20.70
3     6                   Berkshire Hathaway          19.25
4   324                           NRG Energy          15.56
5   478                               NetApp          14.38
6   447                            FM Global          14.36
7    89  Massachusetts Mutual Life Insurance           8.30
8   328                              Newmont           7.23
9   327                           Altice USA           6.38

In [46]:
revenue_change_df = df.sort_values(by = ['revenue_change'], ascending = False, ignore_index=True)
revenue_change_df[['rank','company','revenue_change']][:10]

rank                 company  revenue_change
0    13                   Cigna            2.16
1   390               Andersons            1.68
2   133    United Natural Foods            1.30
3   250   L3Harris Technologies            1.08
4   111               ViacomCBS            0.92
5   388  Westinghouse Air Brake            0.88
6   381       CommScope Holding            0.83
7   349                     KKR            0.75
8   311                  Fiserv            0.75
9   288        Keurig Dr Pepper            0.49

In [37]:
market_value_df = df.sort_values(by = ['market_value'], ascending = False, ignore_index=True)
market_value_df[['rank','company','market_value']][:10]

company  market_value
0           Microsoft     1199549.7
1               Apple     1112640.8
2          Amazon.com      970680.1
3            Alphabet      798905.3
4            Facebook      475455.2
5  Berkshire Hathaway      442897.1
6   Johnson & Johnson      345704.9
7             Walmart      321803.3
8                Visa      316199.1
9      JPMorgan Chase      276750.1

In [42]:
employees_df = df.sort_values(by = ['no_of_employees'], ascending = False, ignore_index=True)
employees_df[['rank','company','no_of_employees']][:10]

rank                company  no_of_employees
0     1                Walmart        2200000.0
1     2             Amazon.com         798000.0
2   361     Yum China Holdings         450000.0
3    23                 Kroger         435000.0
4    26             Home Depot         415700.0
5     6     Berkshire Hathaway         391500.0
6    47                  FedEx         389500.0
7    38                    IBM         383800.0
8    43  United Parcel Service         377640.0
9    37                 Target         368000.0

In [47]:
profit_df[['rank','company', 'profits($millions)']][:10].to_csv('Top 10 Profit.csv')
profit_change_df[['rank','company','profit_change']][:10].to_csv('Top 10 Profit Increased.csv')
profit_change_df[['rank','company','profit_change']][-10:].to_csv('Companies with Most Loss Rate.csv')
revenue_change_df[['rank','company','revenue_change']][:10].to_csv('Top 10 Revenue Increased.csv')
market_value_df[['rank','company','market_value']][:10].to_csv('Top 10 Highest Market Value.csv')
employees_df[['rank','company','no_of_employees']][:10].to_csv('Highest Employees.csv')

In [48]:
employees_df[['rank','company','no_of_employees']][-10:]

rank                company  no_of_employees
490   319                Calpine           2256.0
491   413          DCP Midstream           2250.0
492   100           StoneX Group           2012.0
493   419           Devon Energy           1800.0
494   461                   Hess           1775.0
495   329        Cheniere Energy           1530.0
496   349                    KKR           1384.0
497   483       EnLink Midstream           1355.0
498   457  Brighthouse Financial           1330.0
499   127    NGL Energy Partners           1300.0

## Summary Table :

In [59]:
df2 = df.set_index('company')
df2[:5]

rank  no_of_employees  ...  market_value  rank_is_changed
company                             ...                               
Walmart         1        2200000.0  ...      321803.3            False
Amazon.com      2         798000.0  ...      970680.1             True
Exxon Mobil     3          74900.0  ...      160696.3             True
Apple           4         137000.0  ...     1112640.8             True
CVS Health      5         290000.0  ...       77375.8             True

[5 rows x 10 columns]

In [71]:
summary_list = ['Walmart','Amazon.com','Exxon Mobil','Apple','CVS Health','Berkshire Hathaway',
                'Pfizer','Alphabet','Facebook','Intel','NetApp','Calpine']
df2.loc[summary_list,['rank','rank_change','revenue_change','profit_change','profits($millions)']]

rank  rank_change  ...  profit_change  profits($millions)
company                                ...                                   
Walmart                1          0.0  ...           1.23             14881.0
Amazon.com             2          3.0  ...           0.15             11588.0
Exxon Mobil            3         -1.0  ...          -0.31             14340.0
Apple                  4         -1.0  ...          -0.07             55256.0
CVS Health             5          3.0  ...           0.00              6634.0
Berkshire Hathaway     6         -2.0  ...          19.25             81417.0
Pfizer                64         -3.0  ...           0.46             16273.0
Alphabet              11          4.0  ...           0.12             34343.0
Facebook              46         11.0  ...          -0.16             18485.0
Intel                 45         -2.0  ...           0.00             21048.0
NetApp               478          5.0  ...          14.38              1169.0
Calpine              319         11.0  ...          76.00               770.0

[12 rows x 5 columns]

In [74]:
my_df = df2.loc[summary_list,['rank','rank_change','revenue_change','profit_change','profits($millions)']]

In [76]:
my_df.to_csv('Summary.csv')

## Data Insights Summary : 

1. On Analysing the top 10 companies with highest profits in 2020, we find that though a few companies had a good profit, their rank has dropped 

> > > Eg: Berkshire Hathaway, Apple, Intel, Wells Fargo, Verizon, etc.

2. Berkshire Hathaway, conglomerate with Warren Buffet as CEO has managed to gain the maximum profits despite the pandemic in 2020 and has 6th Rank in Fortune500 list &  4th rank in Profit Change List. It managed to make more profits in 2020 pandemic, compared to 2019. This shows that Berkshire Hathaway is investing very carefully in a calculated way learning from the past experiences. An interesting observation is that though the top Fortune 500 companies managed to top the Profit List, only Berkshire Hathaway has (gained 4th position in Top 10 Profit Increase), had higher profits in 2020 , compared to 2019. 

3. Calpine is an electricity generator company and people spending more time in home & using more current during Pandemic lockdown has resulted in 76% profit increase with just 2256 employees and also jumped 11 positions in the Rank.

4. Walmart, though the 1st rank holder of Fortune500 list is at 14th position with respect to profits and couldn't gain a position in the top 10 Profit_Change Companies.

5. Amazon.com has won the 2nd Rank in the Fortune 500 list, improving from 5th Rank in 2019, due to increase online shopping in pandemic (shop from home), but still at 21st position with respect to profits.

6. Exxon Mobil dropped from Rank 2 in 2019 to Rank 3 in 2020, and saw a decrease in Revenue as well as profits, but managed to get 14th Rank in Top Profit List.

7. Apple's Rank, Revenue, Profit has dropped in 2020 when compared to 2019, probably due to Coronavirus lockdown.But still Apple managed to hold 2nd position with respect to profits 

8. CVS Health has secured 5th Rank in Fortune 500 list & managed to imrove its rank by 3 positions since 2019, most probably due to high medicines & health care demand in 2020 pandemic period.

9. Pfizer is the pharmaceutical company with highest profits and the profit has increased by 0.46M, but its Rank & Revenue is dropped in 2020, inspite of Coronavirus pandemic, a period when lots of medicines are in high demand.

10. Alphabet's rank, revenue & profits have increased due to Coronavirus lockdown as people were at home querying Google, watching Youtube more than ever before. Alphabet has improved its rank by 4 positions 

11. Facebook has increased revenue, but decreased profit and managed to be at the 10th Rank in Companies with highest profits. The use of facebook is increased due to lockdown which improved its rank on Fortune500 list by 11 positions 


12. Microsoft has secured 3rd rank in Profit List, though its at Rank 21 in Fortune500 list(managed to jump 5 ranks above since 2019)

13. Another amazing observation is that Intel has secured 7th position in Top Profit List, indicating the Digital Revolution with increased Online Education, Entertainment, Bank payments, Shopping all of which requires Cloud solutions, Data Centres & Hardwares and also probably the increase in the number of people entering Data field.

14. Net App, a cloud solutions company has obtained 6th Rank in Profit change list, with 14.38 % increase in profits in 2020 compared to 2019.This is also a major sign of the rising Digital Revolution.

15. Most of the companies which have had a higher profit in 2020 , compared to 2019 are the Banking & Financial sectors. This could be probably due to higher loss of jobs in 2020 pandemic, leading to more people borrowing from Banks and online payments , online shopping. Cash less transactions (in fear of Coronavirus spreading through currency notes) & digitalizing the Financial sector has also led to huge profits for the Banks.  




